lib/vanagon/component/dsl.rb

```diff
@@ -289,18 +289,56 @@ def environment(env)
         @component.environment.merge!(env)
       end
 
-      # Adds actions to run at the beginning of a package install
+      # Adds action to run during the preinstall phase of packaging
       #
-      # @param action [String] Bourne shell compatible scriptlets to execute
-      def add_preinstall_action(action)
-        @component.preinstall_actions << action
+      # @param pkg_state [Array] the state in which the scripts should execute. Can be
+      #   one or multiple of 'install' and 'upgrade'.
+      # @param block [Array] the Bourne shell compatible scriptlet(s) to execute
+      def add_preinstall_action(pkg_state, block)
```

Comment:
I think calling the parameter block is going to be confusing later if it isn't expected to actually be a block.